### PR TITLE
Add formula for youtube-dl-gui.

### PR DESCRIPTION
**Pull request by pr0d1r2**

### CODE REVIEW

File: Library/Formula/youtube-dl-gui.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class YoutubeDlGui < Formula
+  homepage "https://github.com/MrS0m30n3/youtube-dl-gui"
+  url "https://github.com/MrS0m30n3/youtube-dl-gui.git",
+    :revision => "36c8147a0ba4da01215d92957246527285cca710"
+
+  # cannot use git tag right now as new version with OSX support not released
```
Comment (pr0d1r2) [inline Library/Formula/youtube-dl-gui.rb]: Asked creator for that: https://github.com/MrS0m30n3/youtube-dl-gui/issues/41

File: Library/Formula/youtube-dl-gui.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    system "python", "setup.py", "install", "--prefix=#{prefix}"
+    system "mkdir", "-p", "#{prefix}/YoutubeDlGui.app/Contents/MacOS"
+    system "cp", "#{prefix}/bin/youtube-dl-gui", "#{prefix}/YoutubeDlGui.app/Contents/MacOS/YoutubeDlGui"
+    system "mkdir", "-p", "#{prefix}/YoutubeDlGui.app/Contents/Resources/"
+    system "cp", "osx/YoutubeDlGui.icns", "#{prefix}/YoutubeDlGui.app/Contents/Resources/YoutubeDlGui.icns"
+    system "cp", "osx/empty.lproj", "#{prefix}/YoutubeDlGui.app/Contents/Resources/empty.lproj"
+    system "cp", "osx/Info.plist", "#{prefix}/YoutubeDlGui.app/Contents/Info.plist"
+    system "cp", "osx/PkgInfo", "#{prefix}/YoutubeDlGui.app/Contents/PkgInfo"
+  end
+end
```
Comment (mikemcquaid) [inline Library/Formula/youtube-dl-gui.rb]: Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/youtube-dl-gui.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  def install
+    ENV.prepend_create_path "PYTHONPATH", "#{lib}/python2.7/site-packages"
+    system "python", "setup.py", "install", "--prefix=#{prefix}"
+    system "mkdir", "-p", "#{prefix}/YoutubeDlGui.app/Contents/MacOS"
+    system "cp", "#{prefix}/bin/youtube-dl-gui", "#{prefix}/YoutubeDlGui.app/Contents/MacOS/YoutubeDlGui"
+    system "mkdir", "-p", "#{prefix}/YoutubeDlGui.app/Contents/Resources/"
+    system "cp", "osx/YoutubeDlGui.icns", "#{prefix}/YoutubeDlGui.app/Contents/Resources/YoutubeDlGui.icns"
+    system "cp", "osx/empty.lproj", "#{prefix}/YoutubeDlGui.app/Contents/Resources/empty.lproj"
+    system "cp", "osx/Info.plist", "#{prefix}/YoutubeDlGui.app/Contents/Info.plist"
+    system "cp", "osx/PkgInfo", "#{prefix}/YoutubeDlGui.app/Contents/PkgInfo"
```
Comment (mikemcquaid) [inline Library/Formula/youtube-dl-gui.rb]: Skip all this creation of a `.app`, please. It should be in the upstream buildsystem if it's needed.

**Comment by DomT4** [comment]
Upstream have made a tag available. Are you still interested here?

**Comment by pr0d1r2** [comment]
@DomT4 i have issue with running it on clean homebrew install: https://github.com/pr0d1r2/homebrew-contrib/issues/1 - any ideas?

**Comment by DomT4** [comment]
`wx` is part of `wxpython` in Homebrew. With that installed it should be able to `import wx` fine. Not sure why it wouldn't work for goozzik - Suspect more a configuration error than something systemic.

**Comment by pr0d1r2** [comment]
@DomT4 I have confirmation from another person that it happens.

**Comment by DomT4** [comment]
Will need something like `brew gist-logs youtube-dl-gui` to be able to look at this properly.

**Comment by goozzik** [comment]
https://gist.github.com/anonymous/2fd16e646882e6275792

**Comment by DomT4** [comment]
@goozzik What happens if you enter into the Python shell and do `import wx`? The Python shell is usually quite verbose, which is useful.

**Comment by goozzik** [comment]
```
$ python
Python 2.7.6 (default, Sep  9 2014, 15:04:36)
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import wx
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named wx
>>>
```

**Comment by DomT4** [comment]
Welp.

Can you do `brew remove wxpython && brew update && brew install wxpython -fs && brew gist-logs wxpython && python -c "import wx"`?

**Comment by goozzik** [comment]
Thanks, it works ok now.

**Comment by DomT4** [comment]
Great! :+1:. @pr0d1r2 If you want to go ahead with the PR here, we'd welcome that.

**Comment by pr0d1r2** [comment]
@DomT4 i suppose that this is it :)

File: Library/Formula/youtube-dl-gui.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
+class YoutubeDlGui < Formula
+  desc "A cross platform front-end GUI of the popular youtube-dl."
+  homepage "https://github.com/MrS0m30n3/youtube-dl-gui"
+  url "https://github.com/MrS0m30n3/youtube-dl-gui.git",
+    :revision => "95189d3d82c159e069697fa65058c8a9f88ff686"
```
Comment (DomT4) [inline Library/Formula/youtube-dl-gui.rb]: You should be able to use the tarball here. Upstream tagged one for you in https://github.com/MrS0m30n3/youtube-dl-gui/releases/tag/0.3.8.

File: Library/Formula/youtube-dl-gui.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+  def install
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
+    %w[altgraph modulegraph macholib py2app].each do |r|
+      resource(r).stage do
+        system "python", *Language::Python.setup_install_args(libexec/"vendor")
+      end
+    end
+
+    ENV.prepend_create_path "PYTHONPATH", "/usr/local/lib/python2.7/site-packages"
```
Comment (DomT4) [inline Library/Formula/youtube-dl-gui.rb]: You shouldn't need this.

File: Library/Formula/youtube-dl-gui.rb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    ENV.prepend_create_path "PYTHONPATH", "#{lib}/python2.7/site-packages"
```
Comment (DomT4) [inline Library/Formula/youtube-dl-gui.rb]: Are we installing something into this path? Ideally if we are it should be `libexec/"lib/python2.7/site-packages"`.

**Comment by DomT4** [comment]
You'll need to rebase this on an up-to-date `master` branch so the PR knows what a `desc` is.

File: Library/Formula/youtube-dl-gui.rb [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    system "python", "setup.py", "py2app"
+    cp_r "dist/Youtube-DLG.app", "#{prefix}/YoutubeDlGui.app"
+  end
+
+  test do
+    contents_path = "#{prefix}/YoutubeDlGui.app/Contents"
```
Comment (DomT4) [inline Library/Formula/youtube-dl-gui.rb]: Meh. If we can't test actual functionality let's not work around the bot failure with a "fake" test. If we must we can ignore the test-bot's flagging of a lack of test here.

Comment (pr0d1r2) [inline Library/Formula/youtube-dl-gui.rb]: What about https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Formula-Cookbook.md#add-a-test-to-the-formula => "However, a bad test is better than no test at all." ?

File: Library/Formula/youtube-dl-gui.rb (hunk 4 above)
Comment (pr0d1r2) [inline Library/Formula/youtube-dl-gui.rb]: @DomT4 released tarball (https://github.com/MrS0m30n3/youtube-dl-gui/releases/tag/0.3.8) does not contain support for py2app installment. @MrS0m30n3 -> can you please make new release with current py2app which was added in https://github.com/MrS0m30n3/youtube-dl-gui/commit/3d8aa8a0f11442fe96ca41fc9fa2d815488b0afc ?

Comment (MrS0m30n3) [inline Library/Formula/youtube-dl-gui.rb]: A new release will be available after closing https://github.com/MrS0m30n3/youtube-dl-gui/issues/39
The time depends on how fast i am going to get a reply on https://github.com/rg3/youtube-dl/issues/6119

File: Library/Formula/youtube-dl-gui.rb (hunk 7 above)
Comment (DomT4) [inline Library/Formula/youtube-dl-gui.rb]: A bad test still has to be an actual functionality test :)

**Comment by DomT4** [comment]
You'll need to rebase this against an up-to-date `master` branch before next push - The PR currently doesn't know what a `desc` is.

**Comment by pr0d1r2** [comment]
@DomT4 rebased & test removed.

**Comment by mikemcquaid** [comment]
@BrewTestBot test this please

**Comment by mikemcquaid** [comment]
@BrewTestBot test this please

**Comment by mikemcquaid** [comment]
This is maybe better shipped as a binary in Homebrew Cask?

**Comment by bfontaine** [comment]
Anything new here?

**Comment by pr0d1r2** [comment]
nope

**Comment by bfontaine** [comment]
You might want to maintain this in your own tap and re-submit a PR later when upstream will tag a release.

**Comment by MrS0m30n3** [comment]
Agree with @bfontaine. I just came back and there are lot of things that needs to be done before making available a new release.

**Comment by DomT4** [comment]
Closing at this point. We'd probably take a resubmission to our `homebrew/gui` tap once the remaining issues are resolved if anyone is interested then, Thanks!

**Pull request closed without merging**